src/ngx_http_modsecurity_module.c

```diff
@@ -727,6 +744,14 @@ ngx_http_modsecurity_merge_conf(ngx_conf_t *cf, void *parent, void *child)
         return strdup(error);
     }
 
+    // This is necessary for logs to be reopened when the config is reloaded
+    if (msc_rules_reopen_audit_log(c->rules_set, &error) < 0) {
+        return strdup(error);
```

Comment:
This is the same thing that is being done a few lines up, but I'm concerned because the memory allocated for `error` in the `ModSecurity` library is never freed. It may not matter though if config errors cause the worker to die. We need to better understand what is going on here.